### ISSUE TITLE
Multicast ipv6 addressing ff02::1%<interface>

### ISSUE THREAD

**Issue by dcomins** [Tanganelli/CoAPthon#114]
Hi,
I have some nodes which work on ipv6 connected to my Pi, i have a python program running with Coapthon and i am able to communicating with them using their unicast ip. Therefore when i want to use multicast coap (with the addres ff02::1%fslthr0, where fslthr0 is my TUN interface) it doesnt work. Any solution? This library allows the use of an ipv6 multicast address with interface (%fslthr0)? 

**Note:** I can ping the nodes using the all addres ff02::1%fslthr0. But if I only ping ff02::1 it doesn work, so i need to put ff02::1%fslthr0 address in coapthon.
Best
Diego C.

**Comment by Tanganelli** [comment]
Can you ping without specifying the interface with unicast addresses?

Il giorno gio 7 giu 2018 alle ore 10:30 dcomins <notifications@github.com>
ha scritto:

**Comment by dcomins** [comment]
Yes, I ping with no problems without specifying the interface with unicast addresses. But for pinging a multicast address in ipv6 it is mandatory to specify the interface, see: https://superuser.com/questions/840767/ipv6-multicast-address-for-all-nodes-on-network

Txthings library also had problems with this multicast issue, see the second comment from "sjlongland": https://github.com/mwasilak/txThings/issues/16

**Comment by dcomins** [comment]
Tanganelli can you fix this issue? I think there are many people interested on it.

**Comment by Tanganelli** [comment]
Yes, I will try to fix it in the next weeks. Sorry but I am really busy at
the moment.

Il giorno lun 11 giu 2018 alle ore 13:41 dcomins <notifications@github.com>
ha scritto:

**Comment by dcomins** [comment]
Ok, I really appreciate it. Thanks in advance.
Diego C.

**Comment by dcomins** [comment]
Have you made some advances Tanganelli?

**Comment by pchwalek** [comment]
I struggled with this issue and found the solution to be binding the socket to my interface. After doing so, multicast sending worked!! =)

(python3) in client\coap.py, for binding to the wpan0 interface, I added:
`self._socket.setsockopt(socket.SOL_SOCKET, 25, str("wpan0" + '\0').encode('utf-8'))`

**Comment by aslehv** [comment]
Is this solved? I am trying to multicast aswell to ff03::1 or ff02::1 :) I could not get, @pchwalek solution to work.
